tests/utils/state_locks_filter.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+import argparse
+import re
+import sys
+from typing import IO
+
+class LockOpTrace:
+    """
+    LockOpTrace: Represents a Lock Operation trace in the log file.
+    It holds the lines for the trace and allows matching against a part of the trace.
+    """
+
+    def __init__(self, lines: list[str]):
+        self.lines = lines
+        self.hash = hash(str(self))
+
+    def get_trace_lines(self) -> list[str]:
+        return self.lines
+
+    def match(self, part: str) -> bool:
+        for line in self.lines:
+            if part in line:
+                return True
+
+        return False
+
+    def __str__(self) -> str:
+        return "".join(self.lines)
+
+    def __hash__(self) -> int:
+        return self.hash
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, LockOpTrace):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
```

Comment:
```suggestion
            return False
```
Perhaps this got lost before. I think the right thing to do here is to return false. The answer to "is this object (that is not the same type as that object) the same as that object?" is false, not an exception. 

```
>>> {} == 42
False
>>> "" == []
False
```